### PR TITLE
feat: what dose -> what does 

### LINKED ISSUE #4315: The `DoesOrDose` linter misses some cases where "Dose" is written instead of "Does"
Image

> What dose the term 'temperature' refer to in the context of ...

Note that the spelling/grammar checker built into Chrome does flag it: 

Image

### PR DESCRIPTION
**Pull request by FranBowden**
# Issues 

Resolves #4315

# Description

Checks for "what dose" mistakes and corrects it to "what does"

# Demo

Production:
image

Localhost:
image

# How Has This Been Tested?

I have tested this by adding two tests in 'one_to_one_tests.rs' and running 

`cargo test -p harper-core corrects_what_dose` &
`cargo test -p harper-core dont_fix_what_dose_false_positive` (although this is ignored, following conventions from the previous tests above) in the CLI 

I also did a general test running `cargo test -p harper-core` and found no errors
I then tested by running `just dev-web` and in the editor, I wrote `what dose this mean` which then suggested a correction to `what does this mean`

I compared this to the production version where nothing was picked up or found regarding the mistake

# AI Disclosure

- [x] I am a human and didn't use any AI.
- [ ] I used LLM features of my editor, but not an agent.
- [ ] I consulted one or more coding AIs, but didn't use an agent.
- [ ] I used an AI agent interactively.
- [ ] I am an agent or I got an agent to do the work autonomously.

# If Your PR Implements or Enhances a Linter

- [x] I made up the sentences in the unit tests.
- [ ] The sentences in the unit tests were generated by an AI.
- [ ] I'm using examples from the bug report / feature request.
- [ ] I collected real-world sentences for the unit tests.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have added tests to cover my changes
- [x] I have considered splitting this into smaller pull requests.

### CODE REVIEW

File: harper-core/src/linting/phrase_set_corrections/mod.rs [hunk 1]
```diff
@@ -200,6 +200,7 @@ pub fn lint_group() -> LintGroup {
                 ("someone dose", "someone does"),
                 // Interrogatives
                 ("how dose", "how does"),
+                ("what dose", "what does"),
```
Comment (hippietrail) [inline harper-core/src/linting/phrase_set_corrections/mod.rs]: This is too simplistic as it was obviously always going to lead to false positives.
Ignored tests for false positives are fine when we expect them to be rare edge cases but this one is going to be common. When unsure a few Google searches will help clarify.

Comment (FranBowden) [reply harper-core/src/linting/phrase_set_corrections/mod.rs]: Oh I see, I've updated it so that "what dose" has a following word to be more specific and to reduce false positives. I added some more unit tests to cover these changes and found some examples via google search

File: harper-core/src/linting/phrase_set_corrections/mod.rs [hunk 2]
```diff
@@ -200,7 +200,10 @@ pub fn lint_group() -> LintGroup {
                 ("someone dose", "someone does"),
                 // Interrogatives
                 ("how dose", "how does"),
-                ("what dose", "what does"),
+                ("what dose it", "what does it"),
+                ("what dose this", "what does this"),
+                ("what dose that", "what does that"),
+                ("what dose the", "what does the"),
```
Comment (hippietrail) [inline harper-core/src/linting/phrase_set_corrections/mod.rs:208]: "a" and "an" should work too.

**Review by hippietrail — CHANGES_REQUESTED** [review]

**Comment by hippietrail** [comment]
Since you're not using a coding AI, try Googling for real-world examples: `what dose it OR this OR that OR the OF a OR an site:github.com` - I always look on GitHub first for examples because Harper targets mistakes in code comments specifically.

**Review by elijah-potter — APPROVED** [review]
